core/src/main/java/com/linecorp/armeria/server/Route.java

```diff
@@ -61,7 +61,35 @@ static RouteBuilder builder() {
      * @see RouteBuilder#matchesHeaders(Iterable)
      * @see RouteBuilder#matchesParams(Iterable)
      */
-    RoutingResult apply(RoutingContext routingCtx);
+    default RoutingResult apply(RoutingContext routingCtx) {
+        return apply(routingCtx, false);
+    }
+
+    /**
+     * Matches the specified {@link RoutingContext} and extracts the path parameters from it if exists.
+     *
+     * @param routingCtx a context to find the {@link HttpService}
+     * @param routeDecoratingService tells whether this method is called in {@link RouteDecoratingService}.
```

Comment:
`RouteDecoratingService` is an implementation detail; you're not supposed to mention it here. Javadoc will not hyperlink.

How about:

```
@param isRouteDecorator {@code true} if this method is called for route decorators. 
                        {@code false} if this method is called for services.
```

Reply:
Thanks fixed. 😉